### PR TITLE
Tweaks to DID Exchange - inviter/invitee references; Attach fields; headings lint cleanup

### PR DESCRIPTION
**Pull request by swcurran**
Some further cleanup on the DID Exchange RFC.

- Some references to the "inviter" and "invitee" were left from my last clean up `#fail`
- Adjusted the `attach~did-doc` field to match the RFC 0017 Attachments format
- Changed some titles to eliminate some markdown lint complaints because such issues bug me

I think a quick review and merge is likely OK with this one.

This does NOT address #534 but the example attachment change was a result of a comment in that issue.

### CODE REVIEW

**Comment by swcurran** [comment]
@dhh1128 @TelegramSam @kdenhartog @llorllale -- please review and if ok, merge.

**Review by kdenhartog — APPROVED** [review]
Looks good to me. I'm +1 to moving from inviter/invitee to requestor/responder

**Review by llorllale — APPROVED** [review]
LGTM although we need to fix the attachments in the other examples as well

File: features/0023-did-exchange/README.md [hunk 1]
```diff
@@ -282,25 +282,29 @@ The exchange response message is used to complete the exchange. This message is
...
-     "base64": "eyJ0eXAiOiJKV1Qi... (bytes omitted)",
-     "jws": {
-        "header": {
-           "kid": "did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"
-        },
-        "protected": "eyJhbGciOiJFZERTQSIsImlhdCI6MTU4Mzg4... (bytes omitted)",
-        "signature": "3dZWsuru7QAVFUCtTd0s7uc1peYEijx4eyt5... (bytes omitted)"
+      "@id": "B.did@B:A",
+      "mime-type": "application/json",
+      "data": {
```
Comment (sklump) [inline features/0023-did-exchange/README.md:287]: Note that there are three more examples in the text that need this treatment (add `"@id"`, `"mime-type"` and subsume the rest into value for `"data"` key)

**Review by sklump — CHANGES_REQUESTED** [review]
Look closely for three more did_doc~attach examples that need to align with RFC 17

**Comment by swcurran** [comment]
Doh!  Thanks George. I'll get to it later today.

File: features/0023-did-exchange/README.md [hunk 2]
```diff
@@ -330,15 +334,15 @@ Possible reasons:
...
+##### Response Processing Error
 
 - unknown processing error
 
 ## 3. Exchange Complete
 
 The exchange `complete` message is used to confirm the exchange to the _responder_. This message is **required** in the flow, as it marks the exchange complete. The _responder_ may then invoke any protocols desired based on the context expressed via the `pthid` in the DID Exchange protocol.
 
-#### Example
+### Complete Message Example
```
Comment (sklump) [inline features/0023-did-exchange/README.md:345]: Could this section add a treatment of errors and corresponding problem report error codes for exhange-complete message processing errors; e.g., no such exchange at responder? In the weird case where the responder gets an completion with bad data, there is no direction here so far on how to proceed.

File: features/0023-did-exchange/README.md [hunk 3]
```diff
@@ -313,13 +317,13 @@ When the message is sent, the _responder_ are now in the `response-sent` state.
 
 #### Response Processing
 
-When the _invitee_ receives the `response` message, they will verify the `change_sig` provided. After validation, they will update their wallet with the new information. If the endpoint was changed, they may wish to execute a Trust Ping to verify that new endpoint.
```
Comment (sklump) [inline features/0023-did-exchange/README.md:316]: This is the first and only mention of `change_sig`: what is it?

File: features/0023-did-exchange/README.md (hunk 2 above)
Comment (swcurran) [reply features/0023-did-exchange/README.md:358]: Note: Not yet added.

File: features/0023-did-exchange/README.md (hunk 3 above)
Comment (swcurran) [reply features/0023-did-exchange/README.md:316]: I removed the reference and attempted to clarify above. Basically, if the responder had initiated the connection via an OOB invitation or a public DID and then switched to a peer DID, the DIDDoc signature should use the DID/key from the OOB or public DID.

File: features/0023-did-exchange/README.md (hunk 2 above)
Comment (swcurran) [reply features/0023-did-exchange/README.md:358]: Added the same error sections to Complete as with the other message type. For now, no reasons added, and only an "unknown processing error".

File: features/0023-did-exchange/README.md [hunk 4]
```diff
@@ -143,19 +143,23 @@ The _requester_ may provision a new DID according to the DID method spec. For a
...
   "did": "B.did@B:A",
   "did_doc~attach": {
-     "base64": "eyJ0eXAiOiJKV1Qi... (bytes omitted)",
-     "jws": {
-        "header": {
-           "kid": "did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"
-        },
-        "protected": "eyJhbGciOiJFZERTQSIsImlhdCI6MTU4Mzg4... (bytes omitted)",
-        "signature": "3dZWsuru7QAVFUCtTd0s7uc1peYEijx4eyt5... (bytes omitted)"
+      "@id": "B.did@B:A",
```
Comment (llorllale) [inline features/0023-did-exchange/README.md:146]: Although this example is non-normative, I don't think it's a good idea to add `@id` and set its value to the DID's ID. That will probably mislead developers.

Comment (swcurran) [reply features/0023-did-exchange/README.md:146]: I was certain on what to put there.  What do you suggest?

Comment (llorllale) [reply features/0023-did-exchange/README.md:146]: Any UUID would suffice.

Comment (swcurran) [reply features/0023-did-exchange/README.md]: Done

**Pull request merged**